### ISSUE TITLE
RTL Editing Doesn't Work at All!

### ISSUE THREAD

**Issue by webrush** [prose/prose#775]
The only thing that happens when you switch the language to Hebrew, is that the document direction changes from left to right, but this is where it stops. 

Nothing else works, you can't select a word or a sentence to edit, line breaks are all over the place.
Once you start a new line, the line above it is no longer editable.

The YML front matter panel is in more bad shape than the text editor as there is a mix of English and Hebrew.

I translated prose.io in its entirety to Hebrew and I hope you'll take the time to fix this issue so people can really start using it. If there is anything I can do to contribute, let me know.

**Comment by AmrEldib** [comment]
Similar problems for Arabic. Let me know how I can help.

**Comment by dereklieu** [comment]
@AmrEldib if you can contribute an Arabic translation, that would be very much appreciated. There are [instructions](https://github.com/prose/prose/blob/master/CONTRIBUTING.md#translating) in `contributing.md`. You'll want to know the [prose transifex site](https://www.transifex.com/tristen/prose/) as well.

@medicart I'm looking into the Hebrew translation.

**Comment by mikemorris** [comment]
Possible these bugs stem from unresolved issues in CodeMirror https://github.com/codemirror/CodeMirror/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+rtl

**Comment by dereklieu** [comment]
@mikemorris yeah I think you're right. Thought it was the transifex translation at first, but that file is completed and deployed.

CodeMirror has a [direction](https://github.com/codemirror/CodeMirror/tree/direction) branch that is trying to address this, but isn't being worked on and is now way behind the master.

Open to suggestions here.

**Comment by AmrEldib** [comment]
@dereklieu I'd love to help with the localization. I've submitted a request to be a member of the team.

@dereklieu @mikemorris would testing the direction branch help merge (and then using it in Prose)? I see at least [one comment](https://github.com/codemirror/CodeMirror/issues/2878#issuecomment-62554396) welcoming testing. Another more recent [comment](https://github.com/codemirror/CodeMirror/issues/2927#issuecomment-94414195) is more specific.
What do you think?

**Comment by dereklieu** [comment]
@AmrEldib I haven't looked into the CodeMirror internals much, so I'd be interested in knowing just how much work this task would involve. 

The version of CodeMirror that Prose uses is outdated as well, and needs an update.

**Comment by mehdisadeghi** [comment]
@dereklieu I just submitted a request to join Persian translation team. However, there seems to be multiple join requests waiting there. Are you looking at transifex join requests at all?

**Comment by mikemorris** [comment]
@mehdisadeghi Hrm, just checked and it looks like something got screwed up in Transifex during a redesign or something - I stopped receiving request notifications, and I'm no longer able to approve requests for some reason. I did however add @dereklieu as a maintainer - hopefully he'll be able to get this sorted out.

**Comment by mehdisadeghi** [comment]
Thanks for the checking @mikemorris. I'll wait for @dereklieu to approve it.

**Comment by dereklieu** [comment]
Hm, also can't approve people. @tristen I think it has to be you (or possibly add me as an admin?). I can't access the "Teams" tabs that's in the Transifex top nav.

**Comment by tristen** [comment]
Hmm @dereklieu I see you as a "maintainer" on prose and I don't see any option to make you an "admin" or even whether I'm an "admin" or not. I've sent a _change ownership_ to your "derrrrrrr" account!

**Comment by nikonikoniko** [comment]
Hello - are there any plans to add RTL support for content editing?

**Comment by nikonikoniko** [comment]
It seems to me that the only thing that needs to be added for successful rtl text editing is 
```
.CodeMirror {
    direction: rtl;
}
``` 
when the direction is rtl.

**Comment by dereklieu** [comment]
@nikonikoniko can you submit a PR? I don't know if this has changed, but prior when we had discussed this issue, CodeMirror didn't really support rtl text editing.

**Comment by nikonikoniko** [comment]
Yes, we can work on that.  Thanks for responding!

**Comment by dereklieu** [comment]
Hey @nikonikoniko best if we can avoid adding an extra button. IMO, the ideal behavior would be, if the user switches from English to Hebrew, the class is applied which flips the CodeMirror editor right to left, and when the user switches back to English (or other left-to-right language) the class is removed.

Happy to provide any additional feedback is helpful.

**Comment by nikonikoniko** [comment]
"switches to hebrew" - does this mean here? https://prose.io/#chooselanguage
for multilingual markdown repositories, this is quite hidden, but ok, we can do that.

**Comment by nikonikoniko** [comment]
(also just checking - there are no arabic interface translations at the moment?)

**Comment by dereklieu** [comment]
I believe so

**Comment by tareefdev** [comment]
Hi there
I'm interested in helping with this issue too; I sent a request to join the Arabic translating team on transfix.

**Comment by nikonikoniko** [comment]
There is now a pull request.  Thanks for taking a look!

**Comment by tareefdev** [comment]
Just a gentle reminder about [my pull request](https://github.com/prose/prose/pull/1081/commits/8c25a85b9b243ba98d0b0ed4615e1228f234981f) which brings RTL support when choosing the Arabic interface.

**Comment by babbitt** [comment]
Howdy folks, brand new maintainer just stepping in to say that I'm working on this! We're not 100% there in terms of functionality, but what @tareefdev has proposed is much better than what we currently have. Just working through git challenges to get their changes merged in. I'm working to try and bring this project back from the dead so bear with me. We should have this issue sorted within the next week.

Additionally, I'm aware that Arabic translation has dropped off the main site. I'm working with other maintainers to get that back, just an issue with Transifex changing it's auth requirements. As soon as I have access I'll restore that functionality.

If anyone has the time to pitch in and help with a more robust RTL editing solution I'm happy to welcome all contributions!

Sorry this slipped into limbo for so long!
شكرا / תודה
